include/cuco/detail/open_addressing/open_addressing_ref_impl.cuh

```diff
@@ -378,9 +378,10 @@ class open_addressing_ref_impl {
   {
     static_assert(cg_size == 1, "Non-CG operation is incompatible with the current probing scheme");
 
-    auto const val    = this->heterogeneous_value(value);
-    auto const key    = this->extract_key(val);
-    auto probing_iter = probing_scheme_(key, storage_ref_.bucket_extent());
+    auto const val      = this->heterogeneous_value(value);
+    auto const key      = this->extract_key(val);
+    auto probing_iter   = probing_scheme_(key, storage_ref_.bucket_extent());
+    auto const init_idx = *probing_iter;
```

Comment:
This could technically be handled inside the probing iterator itself by making this a member of the iterator and after every increment check

```cpp
__host__ __device__ constexpr auto operator++() noexcept
{
  curr_index_ = (curr_index_ + step_size_) % upper_bound_;
  return (curr_index_ == init_index_) ? end() : *this;
}
```

Yes or no?

Reply:
Our probing iterator functions essentially as a counting iterator. The end of an iterator should be determined by the user or the iterator's range, not by the iterator itself. For example, `std::vector` provides both `begin()` and `end()`, but `end()` does not inherently know it is a termination point. I recall that we once explored a probing range design, but I cannot remember why we did not finalize it. I will look into this further, and any follow-up work to transition the counting iterator design to a range-based design will be addressed in a separate PR.